azurelinuxagent/common/protocol/wire.py

```diff
@@ -1109,12 +1107,16 @@ def report_event(self, event_list):
                 continue
             if len(buf[event.providerId] + event_str) >= MAX_EVENT_BUFFER_SIZE:
                 self.send_event(event.providerId, buf[event.providerId])
-                buf[event.providerId] = ""
+                buf[event.providerId] = b''
+                logger.verbose("No of events this request = {0}".format(events_per_request))
+                events_per_request = 0
             buf[event.providerId] = buf[event.providerId] + event_str
+            events_per_request += 1
 
         # Send out all events left in buffer.
         for provider_id in list(buf.keys()):
             if len(buf[provider_id]) > 0:
+                logger.verbose("No of events this request = {0}".format(events_per_request))
```

Comment:
Added extra verbose log to capture the amount of events that we send per request, this might be useful in figuring out telemetry pipeline issues if needed.

Reply:
The main use case is to check on a request level how many events we're sending per request and if we dropped any and how long each log takes to respond (since its a single threaded environment). This gives us insight into how long each request to wireserver telemetry takes to respond and also what's the average number of events per thread. 

That was the main motivation behind adding this. If you guys feel this is an overkill I can remove it.